### PR TITLE
Turret bleeding fix

### PR DESCRIPTION
**Pull request by karols146**
I am not sure that it's a proper fix, waiting for comments.

### CODE REVIEW

File: src/vehicle.cpp [hunk 1]
```diff
@@ -4780,8 +4780,8 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                          name.c_str(), parts[ ret.part ].name().c_str(), ret.target_name.c_str());
             }
         }
-
-        if( part_flag( ret.part, "SHARP" ) ) {
+        monster *mon = dynamic_cast<monster *>(critter); // to blood check
+        if( part_flag( ret.part, "SHARP" ) && mon->made_of( material_id( "flesh" ) ) ) { //bleed only on flesh monster
```
Comment (codemime) [inline src/vehicle.cpp]: Please, don't do that. It's a very ugly hack. I'll provide a generic solution pretty soon.

Comment (karols146) [inline src/vehicle.cpp]: Ok no problem, I am just wait for you solution :+1:

Comment (codemime) [inline src/vehicle.cpp]: Thank you. Sorry, I should have announced it earlier.

Comment (karols146) [inline src/vehicle.cpp]: No problem, I am just curious how you solve it.

**Pull request closed without merging**